operator-wg/whitepaper/052_metacontroller.md

```diff
@@ -0,0 +1,29 @@
+## Metacontroller - Lightweight Kubernetes controllers as a service
+
+[Metacontroller](https://metacontroller.github.io/metacontroller/) is an operator, that makes it easy to write and deploy custom operators.
+
+It introduces two CRD's itself (2021) :
+* [Composite Controller](https://metacontroller.github.io/metacontroller/api/compositecontroller.html) - allowing to write operator triggered by CRD
+* [Decorator Controller](https://metacontroller.github.io/metacontroller/api/decoratorcontroller.html) - allowing to write operator triggered by any kubernetes object (also managed by other operators)
+
+Metacontrollers itself, configured by one of its CRD, will take care of observing cluster state and notify user operator in case any action is needed.
+
+User operator should, having given resource as input, compute desired state of dependent objects.
+
+We call this `lambda controler` pattern (more on this [here](https://metacontroller.github.io/metacontroller/concepts.html#lambda-controller)), as the output is calculated only considering input.
+
+Main advantages of metacontroller :
+* you only need to provide a function (called via webhook) without any boilerplate related to watching kubernetes resources etc
+* function can be written in any language, and exposed via http
+
+
+Main limitations :
+* only certain patterns are possible to implement, mentioned above
+* current architecture relies on single metacontroller in a cluster
+* not aware on any external state, relies entirely on cluster state
+
+... which are planned to be improved in the future.
+
+If you are interested what can be implemented using metacontroller, please visit our [examples](https://metacontroller.github.io/metacontroller/examples.html) page !
```

Comment:
I think we should add such things to the related work section (like where to get additional information or further reading if it's ok for you)

Reply:
@thschue so to 11_Related_Work.md ? It would be hard to find for frameworks, unless it will be done in one place there for all frameworks. I can move it but please advise :)